assets/js/common/OperationModals/SaptuneSolutionOperationModal.jsx

```diff
@@ -54,25 +56,39 @@ function SaptuneSolutionOperationModal({
   const [checked, setChecked] = useState(false);
   const [solution, setSolution] = useState(NOT_SELECTED);
 
-  const availableSolutions = filter(solutions, ({ available }) =>
-    available(isHanaRunning, isAppRunning)
-  );
+  const availableSolutions = pipe(
+    filter(({ available }) => available(isHanaRunning, isAppRunning)),
+    map((solutionOption) => ({
+      ...solutionOption,
+      disabled: solutionOption.value === currentlyApplied,
+    }))
+  )(solutions);
+
+  const resetSolution = () => setSolution(currentlyApplied || NOT_SELECTED);
+
+  useEffect(() => {
+    setChecked(false);
+    resetSolution();
+  }, [currentlyApplied]);
+
+  const resetState = () => {
+    resetSolution();
+    setChecked(false);
+  };
 
   return (
     <OperationModal
       title={title}
       description="Select Saptune tuning solution"
       operationText="Saptune solution"
-      applyDisabled={!checked || solution === NOT_SELECTED}
+      applyDisabled={
+        !checked || solution === NOT_SELECTED || solution === currentlyApplied
+      }
       checked={checked}
       isOpen={isOpen}
       onChecked={() => setChecked((prev) => !prev)}
-      onRequest={() => onRequest(solution)}
-      onCancel={() => {
-        onCancel();
-        setSolution(NOT_SELECTED);
-        setChecked(false);
-      }}
+      onRequest={() => pipe(onRequest, () => setChecked(false))(solution)}
```

Comment:
curiosity:
why do we need a pipe here if the result of one is not used in the next.
cannot we just not use it and avoid its execution?
The same in the `onCancel`